### DISCUSSION TITLE
parameters not grouped by parameter groups, reordering par ensemble

### DISCUSSION THREAD

**Discussion by DarkForceOne** [usgs/pestpp#299] [Q&A] [answered]
I am running PESTPP-IES (version 5.2.7) and do not understand the meaning of the message "parameters not grouped by parameter groups, reordering par ensemble" as all parameters in a pest control file are assigned a certain parameter group. Could you clarify? Thank you!

![image](https://github.com/usgs/pestpp/assets/6100112/77763aa8-66ed-489b-8609-644179801447)

**Comment by jtwhite79** [comment, marked as answer]
That is just telling you the parameters in the control file arent aligned by their respective parameter groups and to draw the parameter realizations "by group" (the default behavior), they need to be, so internally pestpp-ies is reordering the parameters.   That message could probably be silenced...

**Reply by DarkForceOne** [reply]
Thanks. This message never caught my attention until I attempted to run PESTPP-IES with A LOT of pilot points. It took quite a long time to finish the "...drawing by group" step. I actually terminated the program at that step several times, until I decided to let it go.

**Reply by jtwhite79** [reply]
Yeah that will be slow if you have huge numbers of pilot points....might be faster to draw the ensemble the pest utils or pyemu and pass it to ies...
